src/components/Button/index.tsx

```diff
@@ -223,9 +224,14 @@ const Base = (props: ThemedButtonProps) => {
 		plain,
 		active,
 		light,
+		compact,
 		...restProps
 	} = props;
 
+	// Add a method to hide the button label. Here we can just pass an array of 
+	// boolean to decide on each breakpoint whether to show the label or not
+	const shouldCompact = useBreakpoint(compact || [false]);
```

Comment:
```suggestion
	const shouldShowLabel = useBreakpoint(compact || [false]);
```

Reply:
@sradevski in doing so we would have a contradiction in the result since `compact true` will remove the label and we would have `shouldShowLabel true` which should also remove the label.

Reply:
@JSReds that's right, I was thinking of `shouldHideLabel`, but I am fine with it as it is as well, not a big deal.